tests/Feature/Submission/Tests/TestXMLTest.php

```diff
@@ -90,4 +91,61 @@ public function testValidXML(): void
             ],
         ]);
     }
+
+    /**
+     * Test parsing a valid Test.xml file that contains non-UTF-8 characters
+     * in the test output.
+     */
+    public function testNonUTF8Output(): void
+    {
+        $this->submitFiles($this->project->name, [
+            base_path(
+                'tests/Feature/Submission/Tests/data/non_utf8_output.xml'
+            ),
+        ]);
+
+        $this->graphQL('
+            query project($id: ID) {
+              project(id: $id) {
+                builds {
+                  edges {
+                    node {
```

Comment:
You could simplify this by just querying the build by ID directly.  A number of tests throughout the codebase were written before I added the top-level `build(id: _)` field and still access builds by project.  No real difference, other than less indentation.

```
query build($id: ID) {
    build(id: $id) {
        tests {
            ...
```